src/main/java/com/amazon/opendistroforelasticsearch/security/resolver/IndexResolverReplacer.java

```diff
@@ -259,36 +252,13 @@ private Resolved resolveIndexPatterns(final IndicesOptions indicesOptions, final
                     log.debug("No such indices for pattern {}, use raw value", localRequestedPatterns);
                 }
 
-                _indices = new ArrayList<>(localRequestedPatterns.size());
-
-                for (String requestedPattern : localRequestedPatterns) {
-                    _indices.add(resolver.resolveDateMathExpression(requestedPattern));
-                }
-
+                _indices = dateResolvedLocalRequestedPatterns;
             }
 
-            final List<String> _aliases = WildcardMatcher.getMatchAny(localRequestedPatterns.toArray(new String[0]), aliases);
-
-            matchingAllIndices.addAll(_indices);
-
-            if (_aliases.isEmpty()) {
-                matchingIndices.addAll(_indices); //date math resolved?
-            } else {
-
-                if (!_indices.isEmpty()) {
-
-                    for (String al : _aliases) {
-                        Set<String> doubleIndices = lookup.get(al).getIndices().stream().map(a -> a.getIndex().getName()).collect(Collectors.toSet());
-                        _indices.removeAll(doubleIndices);
-                    }
-
-                    matchingIndices.addAll(_indices);
-                }
-            }
+            matchingAllIndices = _indices;
         }
 
-        return new Resolved.Builder(matchingAliases, matchingIndices, matchingAllIndices, null, requestedPatterns0, remoteIndices)
-                /*.addTypes(resolveTypes(request))*/.build();
+        return new Resolved(matchingAliases, matchingAllIndices, Collections.emptySet(), Sets.newHashSet(requestedPatterns0), remoteIndices);
```

Comment:
nit: prefer `new HashSet<>(Arrays.asList())`. `Sets.newHashSet()` is likely to be removed or deprecated in Guava.